build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/AckeePluginKotlin.kt

```diff
@@ -166,6 +173,15 @@ class AckeePluginKotlin : Plugin<Project> {
                         }
                     }
                 }
+
+                /**
+                 * Run "lint$BuildVariant" task before every "assemble$BuildVariant" tasks
+                 */
+                android.applicationVariants.forEach { variant ->
+                    if (variant.buildType.isMinifyEnabled) {
+                        variant.assembleProvider.get().dependsOn(tasks.named("lint${variant.name.capitalize()}"))
```

Comment:
You should not use `Task.dependsOn()` but rather `TaskProvider<Task>.dependsOn()`.

```kotlin
variant.assembleProvider.dependsOn(tasks.named("lint${variant.name.capitalize()}"))
```

The difference is that the latter is lazy and will not be configured unless actually needed by the executed task.

With this set-up, if you were to run a task, say, `app:androidDependencies`, all variants' `assemble` tasks will be configured even though they won't be actually executed, wasting resources.

Reply:
Fixing this only on this line 182 won't have any effect on its own since the `variant.assembleProvider.get()` is also called in two other places above (when setting up `copyAndRename` task and when copying `mapping.txt`).

If any of those places call `get()`, then the task must be configured eagerly anyways so fixing one place won't bring any visible benefits.